Python/traceback.c

```diff
@@ -554,29 +686,53 @@ tb_displayline(PyTracebackObject* tb, PyObject *f, PyObject *filename, int linen
         if (start_col_byte_offset < 0 || end_col_byte_offset < 0) {
             goto done;
         }
+
         // Convert the utf-8 byte offset to the actual character offset so we
         // print the right number of carets.
-        Py_ssize_t start_offset = _PyPegen_byte_offset_to_character_offset(source_line, start_col_byte_offset);
-        Py_ssize_t end_offset = _PyPegen_byte_offset_to_character_offset(source_line, end_col_byte_offset);
+        Py_ssize_t start_offset = (Py_ssize_t)start_col_byte_offset;
+        Py_ssize_t end_offset = (Py_ssize_t)end_col_byte_offset;
 
-        char offset = truncation;
-        while (++offset <= start_offset) {
-            err = PyFile_WriteString(" ", f);
-            if (err < 0) {
-                goto done;
+        if (source_line) {
```

Comment:
Why is this guarded by `source_line`? How we can correctly calculate the offsets if we cannot get the line to calculate the characters? How can tHe code path when this is false and we continue with the bytes be corrent?

Reply:
@isidentical This still applies, could you please check out what's going on here?